modules/core/src/test/java/org/apache/ignite/testframework/junits/logger/GridTestLog4jLogger.java

```diff
@@ -458,6 +458,8 @@ public static void removeAppender(FileAppender a) {
         if (!impl.isTraceEnabled())
             warning("Logging at TRACE level without checking if TRACE level is enabled: " + msg);
 
+            assert impl.isTraceEnabled() : "Logging at TRACE level without checking if TRACE level is enabled: " + msg;
```

Comment:
I think you should not throw an exception.

Reply:
Why?

Reply:
If debug is not enabled, you should throw an error so that when you disable the "ea" option, you do not miss the error.

Reply:
"-ea" is enabled on TeamCity runs and IDE test configuration. Anyway according to your logic user can set other logger implementation and avoid those checks.

Reply:
Yes, user can change logger, I believe that in our test framework, we should not use "assert".